### DISCUSSION TITLE
Dependency on `glibc` on Mac?

### DISCUSSION THREAD

**Discussion by peterbecich** [srid/haskell-flake#172] [Q&A] [answered]
I am seeing an error which may or may not relate to `haskell-flake`. I am using an outdated `haskell-flake` from April 2023.

Using `haskell-flake` on a Mac, does `nix flake check` have a dependency on `glibc`? I see `glibc` is not available on Mac: https://search.nixos.org/packages?channel=unstable&show=glibc&from=0&size=50&sort=relevance&type=packages&query=glibc

Here is the Flake: https://github.com/haskell/hackage-server/blob/master/flake.nix

I don't have a Mac handy at the moment. I am seeing this error on a Mac GitHub Action:
```
error: Package ‘glibc-2.37-8’ in /nix/store/22z4n4mxs2vz3l3lg41dz3mgnq1d4wxs-source/pkgs/development/libraries/glibc/default.nix:166 
is not available on the requested hostPlatform:

        hostPlatform.config = "x86_64-apple-darwin"
         package.meta.platforms = [
           "aarch64-linux"
           "armv5tel-linux"
           "armv6l-linux"
           "armv7a-linux"
           "armv7l-linux"
           "i686-linux"
           "m68k-linux"
           "microblaze-linux"
           "microblazeel-linux"
           "mipsel-linux"
           "mips64el-linux"
           "powerpc64-linux"
           "powerpc64le-linux"
           "riscv32-linux"
           "riscv64-linux"
           "s390-linux"
           "s390x-linux"
           "x86_64-linux"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.
```
https://github.com/haskell/hackage-server/pull/1219#issuecomment-1596576063

Thanks for any insights

**Comment by srid** [comment, marked as answer]
One thing I noticed is that your flake doesn't output any `check`, but `nix flake check` still fails.

Apparently `nix flake check` evaluates the devShell. And you have some Linux-only tools in there.

https://github.com/haskell/hackage-server/pull/1219#issuecomment-1597140858

**Reply by peterbecich** [reply]
Good catch, thanks.  Looking for other Linux only tools now: https://github.com/srid/haskell-flake/discussions/172#discussioncomment-6828552

**Reply by peterbecich** [reply]
`cryptodev` was the remaining Linux-only dependency, thx

**Comment by roberth** [comment]
If `--show-trace` doesn't tell you which attribute it evaluated, that'd be a Nix bug.
(Although I'll be the first to admit that its verbosity is not great, it usually contains crucial info)

**Reply by peterbecich** [reply]
Good idea, thanks. Here is the output of `nix flake check --show-trace`, which checks `nix develop`: https://github.com/haskell/hackage-server/actions/runs/5983156055/job/16232993518?pr=1248 I am reading this carefully now.


Here is the `devShell` configuration: https://github.com/peterbecich/hackage-server/blob/bf4c4000bcc8da1cb28fd5a70c57ca69de2efcee/flake.nix#L33-L59

**Reply by peterbecich** [reply]
I think the problem was `cryptodev`. I missed this one earlier. No support for Darwin: https://search.nixos.org/packages?channel=unstable&show=cryptodev&from=0&size=50&sort=relevance&type=packages&query=cryptodev
